shiny/ui/_theme.py

```diff
@@ -0,0 +1,506 @@
+            `.add_rules(primary_color="#ff0000")` is equivalent to
+            `.add_rules("$primary-color: #ff0000;")`.
+        """
+        rules = self._combine_args_kwargs(*args, kwargs=kwargs)
+        self._rules.extend(rules)
+        self._reset_css()
+        return self
+
+    def to_sass(self) -> str:
+        """
+        Returns the custom theme as a single Sass string.
+
+        Returns
+        -------
+        :
+            The custom theme as a single Sass string.
+        """
+        path_functions = path_pkg_preset(self._preset, "_01_functions.scss")
+        path_defaults = path_pkg_preset(self._preset, "_02_defaults.scss")
+        path_mixins = path_pkg_preset(self._preset, "_03_mixins.scss")
+        path_rules = path_pkg_preset(self._preset, "_04_rules.scss")
+
+        sass_lines = [
+            f'@import "{path_functions}";',
+            *self._functions,
+            *self._defaults,
+            f'@import "{path_defaults}";',
+            f'@import "{path_mixins}";',
+            *self._mixins,
+            f'@import "{path_rules}";',
+            *self._rules,
+        ]
+
+        return "\n".join(sass_lines)
+
+    def to_css(
+        self,
+        compile_args: Optional[SassCompileArgs] = None,
+    ) -> str:
+        """
+        Compile the theme to CSS and return the result as a string.
+
+        Returns
+        -------
+        :
+            The compiled CSS for the theme. The value is cached such that previously
+            compiled themes are returned immediately. Adding additional custom Sass code
+            or changing the preset will invalidate the cache.
+        """
+        if self._css:
+            return copy.copy(self._css)
+
+        if self._can_use_precompiled():
+            self._css = self._read_precompiled_css()
+            return copy.copy(self._css)
+
+        check_libsass_installed()
+        import sass
+
+        if compile_args is None:
+            compile_args = {"output_style": "compressed"}
+
+        self._css = cast(
+            str,
+            sass.compile(
+                string=self.to_sass(),
+                include_paths=self._include_paths,
+                **compile_args,  # type: ignore
+            ),
+        )
+
+        return copy.copy(self._css)
+
+    def _read_precompiled_css(self) -> str:
+        path = path_pkg_preset(self._preset, "preset.min.css")
+        with open(path, "r") as f:
+            return f.read()
+
+    def _html_dependency_precompiled(self) -> HTMLDependency:
+        return HTMLDependency(
+            name=f"shiny-theme-{self._preset}",
+            version=self._version,
+            source={
+                "package": "shiny",
+                "subdir": f"www/shared/sass/preset/{self._preset}",
+            },
+            stylesheet={"href": "preset.min.css"},
+            all_files=False,
+        )
+
+    def _html_dependency(self) -> HTMLDependency:
```

Comment:
You could also have a `tagify()` method that just throws and tell you to provide `Theme` to `theme` arg

Reply:
```
from shiny import ui

ui = ui.page_bootstrap(ui.Theme())
```

```
SyntaxError: The `Theme` class is not meant to be used as a standalone HTML tag. Instead, pass the `Theme` object directly to the `theme` argument of any Shiny page function.
```

Reply:
Thanks!